src/Cms/FileRules.php

```diff
@@ -80,8 +80,21 @@ public static function changeSort(File $file, int $sort): bool
      */
     public static function create(File $file, BaseFile $upload): bool
     {
+        // first we want to ensure that we are not trying to
+        // create a duplicate file. So if a file
+        // with the same name already exists
         if ($file->exists() === true) {
-            if ($file->sha1() !== $upload->sha1()) {
+            // $file will be based on the props of the new file,
+            // so we need to get the props of the existing file
+            $existing = kirby()->file($file->id());
```

Comment:
If somebody knows a better way for this, would be great

Reply:
Accessing the file via the parent is faster (microseconds 😅) than accessing the Kirby instance. I'm not sure which one more stable.

````php
$existing = $file->parent()->file($file->filename());
````